scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -0,0 +1,208 @@
+
+public class AnimateImagesWithImageOverlayController {
+
+  @FXML
+  private SceneView sceneView;
+  @FXML
+  private Button controlAnimationButton;
+  @FXML
+  private Slider opacitySlider;
+  @FXML
+  private ComboBox<String> framesComboBox;
+
+  private List<ImageFrame> imageFrames;
+
+  private Integer imageIndex = 0;
+  private Integer period = 67;
+
+  private Timer timer = null;
+  private boolean isTimerRunning = true;
+
+  public void initialize() {
+
+    try {
+
+      // create a new ArcGISScene and set it to the scene view
+      ArcGISScene scene = new ArcGISScene();
+      sceneView.setArcGISScene(scene);
+
+      // create a camera, looking at the pacific southwest sector
+      Point observationPoint = new Point(-116.621, 24.7773, 856977.0);
+      Camera camera = new Camera(observationPoint, 353.994, 48.5495, 0.0);
+
+      // create an envelope of the pacific southwest sector for displaying the image frame
+      Point pointForImageFrame = new Point(-120.0724273439448, 35.131016955536694, SpatialReferences.getWgs84());
+      Envelope imageFrameEnvelope = new Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488);
+      scene.setInitialViewpoint(new Viewpoint(imageFrameEnvelope, camera));
+
+      // create a new tiled layer from the World Dark Gray Base REST service and set it as the scene's basemap
+      Basemap basemap = new Basemap(new ArcGISTiledLayer("https://services.arcgisonline" +
+        ".com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer"));
+      scene.setBasemap(basemap);
+
+      // create a new elevation source from the Terrain3D REST service and set it as the scene's base surface
+      Surface surface = new Surface();
+      surface.getElevationSources().add(new ArcGISTiledElevationSource("https://elevation3d.arcgis" +
+        ".com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"));
+      scene.setBaseSurface(surface);
+
+      // create and append an image overlay to the scene view
+      sceneView.getImageOverlays().add(new ImageOverlay());
+
+      // instantiate a new empty list to hold image frames
+      imageFrames = new ArrayList<>();
+      // get the image files from local storage as an unordered list
+      File[] imageFiles = new File(System.getProperty("data.dir"), "./samples-data/PacificSouthWest").listFiles();
+      // sort the list of image files
+      if (imageFiles != null) {
+        Arrays.sort(imageFiles);
+        // create an image with the given path and use it to create an image frame
+        for (File file : imageFiles) {
+          ImageFrame imageFrame = new ImageFrame(file.getAbsolutePath(), imageFrameEnvelope);
+          imageFrames.add(imageFrame);
+        }
+      }
+
+      // populate the frames combo box with values
+      framesComboBox.getItems().addAll("60 frames per second", "30 frames per second", "15 frames per second");
+      // open the sample at 15fps
+      framesComboBox.getSelectionModel().select(2);
```

Comment:
This is UI stuff so perhaps it's better to do it before we get to setting up the Scene/SceneView. It might be even possible to define all these things in the FXML file, might be worth having a look?
I also think it's counterintuitive to have the high values at the top rather than the bottom.

Reply:
I spent a while researching about how to define the above in the FXML but had no success :( if you know of a way could you let me know please?

Reply:
I don't know whether this is possible off the top of my head, it was just a passing thought! If you couldn't find any way then I doubt I'll find anything either, so no worries leaving it like this